nixos/modules/services/system/self-deploy.nix

```diff
@@ -150,14 +150,14 @@ in
           git init ${repositoryDirectory}
         fi
 
-        ${gitWithRepo} fetch ${lib.escapeShellArg cfg.repository} ${lib.escapeShellArg cfg.branch}
+        ${gitWithRepo} fetch ${cfg.repository} ${lib.escapeShellArg cfg.branch}
```

Comment:
This line doesn't seem necessary to fix the boot loop problem. Escaping seems like a good thing.